### PR TITLE
Added Z-test to one sample t-test

### PR DESCRIPTION
**Pull request by Kucharssim**
Added Z-test to the One Sample T-test options. #2142 

Changes:
- ttestonesample.R
    enabling new input, structure output for the z-test (the main table), footnotes
- ttestonesampleform.ui
    adding two widgets: checkbox button for selecting Z-test, popup window for specifying std. deviation
- TTestOneSample.json
    creating the two new widgets

Consulted with @AlexanderLyNL

### CODE REVIEW

**Comment by akashrajkn** [comment]
@Kucharssim can you please modify the unittest for one sample ttest.

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 1]
```diff
@@ -168,11 +207,45 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
...
-  	               effect size is given by the matched rank biserial correlation.")
-		} 
+		
+		# preparing footnote - if one of the method is selected; or general if more selected
+		if(wantsWilcox && onlyTest){
+		  textEffect <- "For the Wilcoxon test, effect size is given by the matched rank biserial correlation"
+		} else if(wantsStudents && onlyTest){
+		  textEffect <- "For the Student t-test, effect size is given by Cohen's <em>d</em>."
+		} else if(wantsZtest && onlyTest){
+		  textEffect <- "For the Z test, effect size is given by Cohen's <em>d</em> (based on population deviation)."
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestonesample.R]: on the provided population standard deviation

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 2]
```diff
@@ -168,11 +207,45 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
...
+		} else {
+		  textEffect <- "For the Student t-test, effect size is given by Cohen's <em>d</em>; 
+		  for the Wilcoxon test, effect size is given by the matched rank biserial correlation; 
+		  for the Z test, effect size is given by Cohen's <em>d</em> (based on population deviation)."
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestonesample.R]: on the provided population standard deviation

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 3]
```diff
@@ -168,11 +207,45 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
...
+		}
+		
+		# adding a footnote (do not view methods which are not selected)
+		if (!wantsWilcox && !wantsStudents && !wantsZtest) {
+		} else if(wantsWilcox && wantsStudents && !wantsZtest){
+		  .addFootnote(footnotes, symbol = "<em>Note.</em>",
+		               text = gsub(pattern = "; 
+		  for the Z test, effect size is given by Cohen's <em>d</em> \\(based on population deviation\\)",
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestonesample.R]: on the provided population standard deviation

Why the extra \\ here?

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 4]
```diff
@@ -168,11 +207,45 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
...
+		                           replacement = "",
+		                           textEffect)
+		  )
+		} else if(wantsWilcox && !wantsStudents && wantsZtest){
+		  .addFootnote(footnotes, symbol = "<em>Note.</em>",
+		               text = gsub(pattern = "For the Student t-test, effect size is given by Cohen's <em>d</em>; 
+		  f",
+		                           replacement = "F",
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestonesample.R]: replacements = "F" ?

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 5]
```diff
@@ -146,11 +153,43 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
...
+		} else if(wantsZtest && onlyTest){
+		  textDifference <- "For the Z-test, location parameter 
+		  is given by mean difference <em>d</em>."
+		} else {
+		  textDifference <- "For the Student t-test and Z-test, location parameter 
+is given by mean difference <em>d</em>; for the Wilcoxon test, effect size is given by the Hodges-Lehmann estimate."
+		}
+		
+		# adding a footnote (do not view methods which are not selected)
+		if (!wantsWilcox && !wantsStudents && !wantsZtest) {
```
Comment (akashrajkn) [inline JASP-Engine/JASP/R/ttestonesample.R]: The first `if` condition is empty, can you modify (or remove) it?

File: JASP-Engine/JASP/R/ttestonesample.R (hunk 4 above)
Comment (akashrajkn) [reply JASP-Engine/JASP/R/ttestonesample.R]: I think `replacement = "F"` is correct.

**Comment by boutinb** [comment]
Hoi Simon,

The unit test for ttestonesample fails (see details of the Travis CI build). This file is in JASP-Tests/R/tests/testthat.
In order to run the unit test in R Studio:
library(jasptools, lib.loc="/Users/simon/JASP/jasp-dektop/Tools/")
jasptools::testAll()
or jasptools::testAnalysis("TTestOneSample") to run only these tests.
You may add a new test in case.

**Pull request merged**